experimental/AlgebraicStatistics/src/Markov.jl

```diff
@@ -31,32 +27,31 @@ with the Macaulay2 package `GraphicalModels`.
 ## Examples
 
 ```jldoctest
-julia> R = markov_ring("A" => 1:2, "B" => 1:2, "X" => 1:2, "Y" => 1:2)
-MarkovRing for random variables A -> {1, 2}, B -> {1, 2}, X -> {1, 2}, Y -> {1, 2} in 16 variables over Rational field
+julia> R = markov_ring(2,2,2,2)
+MarkovRing over Rational field for 4 random variables and states (2, 2, 2, 2)
 ```
 """
-function markov_ring(rvs::Pair{<:VarName, <:AbstractArray}...; unknown::VarName="p", K::Field=QQ, cached=false)
-  random_variables = [p.first for p in rvs];
-  state_spaces = [p.second for p in rvs];
+function markov_ring(F::Field, states::Int...; unknown::VarName="p", cached=false)
+  @req length(states) >= 1 "need at least one random variable"
+  @req all(s -> s >= 1, states) "all state spaces must have at least one element"
+  random_variables = 1:length(states);
+  state_spaces = [1:s for s in states];
   varindices = collect(Iterators.product(state_spaces...))
   varnames = [["$(unknown)[$(join(i, ", "))]" for i in varindices]...]
-  R, p = polynomial_ring(K, varnames; cached=cached)
+  R, p = polynomial_ring(F, varnames; cached=cached)
   d = Dict([varindices[i] => p[i] for i in 1:length(varindices)])
-  return MarkovRing(
-    R,
-    d,
-    random_variables,
-    state_spaces
-  )
+  return MarkovRing(R, d, state_spaces)
 end
 
+markov_ring(states::Int...; unknown::VarName="p", cached=false) =
+  markov_ring(QQ, states...; unknown=unknown, cached=cached)
+
 const tensor_ring = markov_ring
 
 function Base.show(io::IO, R::MarkovRing)
-  arrow = Oscar.is_unicode_allowed() ? "→" : "->"
-  print(io, "$(typeof(R)) for random variables ",
-    join([string(x) * " $(arrow) " * "{" * join(R.state_spaces[i], ", ") * "}" for (i, x) in Iterators.enumerate(R.random_variables)], ", "),
-    " in $(length(gens(ring(R)))) variables over $(base_ring(ring(R)))"
+  coeffs = base_ring(R.ring)
+  print(io, "$(typeof(R)) over $(coeffs) for $(length(R.state_spaces)) random variables and states ",
```

Comment:
```suggestion
  io = pretty(io)
  print(io, "Markov ring over ", Lowercase(), base_ring(R.ring), " for $(length(R.state_spaces)) random variables and states ",
```
and then of course you need to fix a bunch of doctest outputs (resp. let Oscar do that fixing for you)